ledger/participant-state/kvutils/src/main/scala/com/daml/ledger/participant/state/kvutils/OffsetBuilder.scala

```diff
@@ -4,72 +4,101 @@
 package com.daml.ledger.participant.state.kvutils
 
 import java.io.{ByteArrayInputStream, ByteArrayOutputStream, DataInputStream, DataOutputStream}
-
 import com.daml.ledger.offset.Offset
 
+import java.nio.ByteBuffer
+
 /** Helper functions for generating 16 byte [[Offset]]s from integers.
   * The created offset will look as follows:
-  * | highest index (64 bits) | middle index (32 bits) | lowest index (32 bits) |
+  * | version (8 bits) | highest index (56 bits) | middle index (32 bits) | lowest index (32 bits) |
   * Leading zeros will be retained when generating the resulting offset bytes.
   *
   * Example usage:
   *
-  *   - If you have one record per block then use [[OffsetBuilder.fromLong]] with a single argument, the block ID.
-  *   - If you may have multiple records per block then use [[OffsetBuilder.fromLong]] with the index within the block as the second argument.
+  *   - If you have one record per block then use [[OffsetBuilder.of]] with a single argument, the block ID.
+  *   - If you may have multiple records per block then use [[OffsetBuilder.of]] with the index within the block as the second argument.
   *
   * @see com.daml.ledger.offset.Offset
   * @see com.daml.ledger.participant.state.kvutils.api.KeyValueParticipantStateReader
   */
 object OffsetBuilder {
-  private[kvutils] val highestStart = 0
+  val DefaultVersion: Byte = 0
```

Comment:
What do you propose?

Reply:
I think we went with a `vN' package and a `Version` constant somewhere else and I think it makes sense. It would also conveniently (as we're going to have another one) make this builder the first version.